### PR TITLE
Extend the frontend 

### PR DESCRIPTION
**Pull request by ghjeong12**
To support integration with Marvel, I update CompoundConfig and fixed a bug.

### CODE REVIEW

File: src/compound-config/compound-config.cpp [hunk 1]
```diff
@@ -519,6 +519,10 @@ bool CompoundConfigNode::getMapKeys(std::vector<std::string> &mapKeys) {
     return false;
   }
 }
+
```
Comment (vmiheer) [inline src/compound-config/compound-config.cpp:532]: It's good to remove extra whitespaces.

File: src/compound-config/compound-config.cpp [hunk 2]
```diff
@@ -533,13 +537,39 @@ CompoundConfig::CompoundConfig(const char* inputFile) {
     YConfig = YAML::Load(f);
     root = CompoundConfigNode(nullptr, YConfig, this);
     useLConfig = false;
+    f.close();
+    std::cout << "file closed" << std::endl;
```
Comment (vmiheer) [inline src/compound-config/compound-config.cpp:551]: Not sure if it's a good idea to have these prints.

File: src/mapping/parser.cpp [hunk 3]
```diff
@@ -129,7 +131,7 @@ Mapping ParseAndConstruct(config::CompoundConfigNode config,
   // Construct the mapping from the parsed sub-structures.
   // A set of subnests, one for each tiling level.
   loop::NestConfig subnests(arch_props_.TilingLevels());
-
+
```
Comment (vmiheer) [inline src/mapping/parser.cpp:142]: It would be great to not have trailing whitespace.

File: src/compound-config/compound-config.cpp (hunk 1 above)
Comment (ghjeong12) [reply src/compound-config/compound-config.cpp:532]: Thanks, will fix it.

File: src/compound-config/compound-config.cpp (hunk 2 above)
Comment (ghjeong12) [reply src/compound-config/compound-config.cpp:551]: I put them for debugging, should have deleted them. Thanks for pointing that out.

**Review by angshuman-parashar — CHANGES_REQUESTED** [review]
- Please change "cycle" to "cycles".
- As far as possible please stick to the coding conventions used in each file. In this specific case, please move the brace in "struct Stats{" to the next line.

**Pull request merged**